internal/encoding/gzip.go

```diff
@@ -0,0 +1,97 @@
+package encoding
+
+import (
+	"compress/gzip"
+	"io"
+	"net/http"
+	"strings"
+)
+
+func GZIPMiddleware(h http.Handler) http.Handler {
+	inner := func(w http.ResponseWriter, r *http.Request) {
```

Comment:
видел похожее решение...
зачем задавать переменную, если можно сразу вернуть функцию
```suggestion
	return func(w http.ResponseWriter, r *http.Request) {
```